shopelectro/views/catalog.py

```diff
@@ -25,11 +25,38 @@ def get_products_count(request):
     return PRODUCTS_ON_PAGE_MOB if mobile_view else PRODUCTS_ON_PAGE_PC
 
 
-def get_paginated_page_or_404(objects, per_page, page_number):
-    try:
-        return Paginator(objects, per_page).page(page_number)
-    except InvalidPage:
-        raise http.Http404('Page does not exist')
+class PaginatedCatalog:
+
+    def __init__(self, number, path, paginated):
+        self.paginated = paginated
+        self.number = number
+        self.index = number - 1
+        self.path = path
+        self.neighbor_bounds = settings.PAGINATION_NEIGHBORS_NUMBER
```

Comment:
with `PAGINATION_NEIGHBORS_NUMBER = 2` we'll get 4 neighbor links: two as `prev_numbers` and two as `next_numbers`.
It can be confusing.
I propose one of this variants:
- call this option `PAGINATION_NEIGHBORS_TOTAL_NUMBER`
- add comment about this option in base.py
- do `self.neighbor_bounds = settings.PAGINATION_NEIGHBORS_NUMBER // 2` here